src/stream/iterator.ts

```diff
@@ -322,25 +315,24 @@ export class OffsetIterator<T = Music21Object> extends _StreamIteratorBase<T> {
             const yieldEls = [e];
             const eOffset = this.srcStream.elementOffset(e);
 
-            for (let forwardIndex = this.index; forwardIndex < this.streamLength; forwardIndex++) {
-                let nextE;
-                try {
-                    nextE = this.srcStreamElements[this.index - 1];
-                } catch (exc) {
+            // allow forwardIndex == this.streamLength because this.index needs
+            // to be incremented inside the loop before being decremented by updateActiveInformation
+            // (when yielding whatever is in yieldEls)
+            for (let forwardIndex = this.index; forwardIndex <= this.streamLength; forwardIndex++) {
+                this.index = forwardIndex;
+                const nextE = this.srcStreamElements[this.index];
+                if (nextE === undefined) {
                     continue;
                 }
                 const nextOffset = this.srcStream.elementOffset(nextE);
                 if (nextOffset !== eOffset) {
-                    this.nextToYield = [nextE];
-                    this.nextOffsetToYield = nextOffset;
                     break;
                 }
                 if (!this.matchesFilters(nextE)) {
                     continue;
                 }
 
-                yieldEls.push(e);
-                this.index = forwardIndex;
```

Comment:
By setting `this.index = forwardIndex` at the bottom of the loop we were getting double runs of the first value for `this.index`. You were right it needed to be placed _somewhere_, though. (I didn't know what updateActiveInformation() was up to when I first looked, so didn't really know what values the indexes were supposed to take.)